ruby/ext/google/protobuf_c/message.c

```diff
@@ -108,7 +108,9 @@ void Message_InitPtr(VALUE self_, upb_Message* msg, VALUE arena) {
   Message* self = ruby_to_Message(self_);
   self->msg = msg;
   RB_OBJ_WRITE(self_, &self->arena, arena);
-  ObjectCache_Add(msg, self_);
+  VALUE stored = ObjectCache_GetSet(msg, self_);
+  (void)stored;
```

Comment:
it prevents a compiler warning when assertions are disabled.

Reply:
Is the assert always valid? Could `ObjectCache_GetSet()` assert that its return value is always its second argument just before returning it? That would obviate the need to have this weird construction to avoid the warning (since `ObjectCache_GetSet()` would actually use the value to return it) and simplify all the call sites by relieving callers of the duty to assert.

Reply:
not in all contexts.  When there are racing calls to `TryAdd` (formerly `GetSet`) with a particular key one of them is guaranteed to win and all of them will return the `VALUE` of that winner.  When other parts of the code knows there cannot be racing calls (as in new object construction), the assert is valid.  When there can be racing calls it is not valid.